### PR TITLE
London | 25-ITP-May | Houssam Lahlah "Person B" | Sprint 3 |  Feature board printer

### PR DESCRIPTION
**Pull request by HoussamLh**
**_Self checklist_**

[x]I have committed my files one by one, on purpose, and for a reason
[x]I have titled my PR with Region | Cohort | FirstName LastName | Sprint | Assignment Title
[x]I have tested my changes
[x]My changes follow the [style guide](https://curriculum.codeyourfuture.io/guides/reviewing/style-guide/)
[x]My changes meet the [requirements](https://github.com/CodeYourFuture/Module-Data-Groups/pull/README.md) of this task

**_## Changelist_**

Description:
- Implemented the functions to display the Tic-Tac-Toe board (printBoard) and to check if no moves are left (checkIfNoMovesLeft). 
- The board output is formatted clearly in the terminal.
- Tested with example boards to verify correct display and detection of full boards.

**_Questions for Reviewer_**

- Is the board output format clear and easy to read in the terminal?

- Is the logic for detecting no remaining moves efficient and correct?

- Any suggestions to improve code readability or style?

### CODE REVIEW

File: __tests__/board-printer.test.js [hunk 1]
```diff
@@ -1,6 +1,21 @@
 import { checkIfNoMovesLeft } from "../board-printer";
 
-test("checkIfNoMovesLeft should return true if there are no moves left", () => {
```
Comment (HoussamLh) [inline __tests__/board-printer.test.js:3]: Syntax Error – Unclosed test() block
The first test() call is missing a closing brace } and is not inside the describe() block. It should either be removed or correctly closed and moved.

File: board-printer.js
Comment (HoussamLh) [inline board-printer.js:1]: Great job on this module! 🎉

printBoard(board)

- The board is displayed clearly and consistently in the terminal.

- Good use of map and join for clean formatting.

- You could consider adding row and column numbers to make it easier for players to choose positions during gameplay.

- Extracting the separator line (=================) into a constant might help if you ever want to change the board style later.

checkIfNoMovesLeft(board)

- The function works as expected and uses clean, efficient logic.

Questions for you:

Have you tested the output visually with partially filled boards and at the end of a game?

Would you consider adding coordinate labels to the printed board to improve the user experience?

File: __tests__/board-printer.test.js
Comment (HoussamLh) [inline __tests__/board-printer.test.js:1]: - Thanks for adding unit tests for checkIfNoMovesLeft()! You've covered both key scenarios:

- A full board returns true

- A board with at least one '_' returns false

However, there are a few issues in the test file structure that need to be addressed:

- Suggestions for Improvement:

Syntax Error – Unclosed test() block
The first test() call is missing a closing brace } and is not inside the describe() block. It should either be removed or correctly closed and moved.

Better Test Naming
You're already doing well here, but make sure your test names clearly describe what is being tested and why. Your describe() and test blocks are appropriately named — good job.

Consider adding one more test case:

test("returns false when only one '_' is left", () => {
  const board = [
    ['X', 'O', 'X'],
    ['O', '_', 'O'],
    ['X', 'O', 'X']
  ];
  expect(checkIfNoMovesLeft(board)).toBe(false);
});

- This helps ensure edge cases are covered (like when only one move remains).

**Comment by HoussamLh** [comment]
Thank you for the review and comments.

**_- I’ve fixed the test file:_**

- Removed the invalid test case at the top.
- Ensured all tests are inside the describe block.
- Added an edge case test when only one _ is left on the board.

Thanks again!
